src/Data/Void.purs

```diff
@@ -2,11 +2,23 @@ module Data.Void (Void, absurd) where
 
 import Data.Show (class Show)
 
--- | An uninhabited data type.
+-- | An uninhabited data type. In other words, one can never create
+-- | a runtime value of type `Void` becaue no such value exists.
 -- |
 -- | `Void` is useful to eliminate the possibility of a value being created.
 -- | For example, a value of type `Either Void Boolean` can never have
 -- | a Left value created in PureScript.
+-- |
+-- | This should not be confused with the word, `void,` that commonly appears in
+-- | C-deriving languages, such as Java:
```

Comment:
nit: I think "C-derived" would be more grammatically appropriate than "C-deriving" here. Or you could use "C-family"? Apparently that's more common now: https://books.google.com/ngrams/graph?corpus=26&smoothing=3&year_start=1800&year_end=2019&content=C-derived+languages%2CC-family+languages&direct_url=t1%3B%2CC%20-%20derived%20languages%3B%2Cc0%3B.t1%3B%2CC%20-%20family%20languages%3B%2Cc0

Reply:
I used `C-family`